### PR TITLE
MNT Build fixes

### PR DESCRIPTION
**Pull request by riedgar-ms**
## Description

Various tweaks to fix the builds:

- Temporary upper bound on `scikit-learn`. Issue #1350 for tracking. A full fix will be required in due course
- Tweaks a couple of the grid generator tests to avoid warnings from `pandas`
- Adjust `_disaggregated_result.py` to cope with recently introduced (and deprecated) argument from `pandas`

Both of the `pandas` changes should fall under #1299 

## Tests

- [ ] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Comment by adrinjalali** [comment]
If we release with this higher bound, users will end up with the highest scikit-learn and the previous version of fairlearn which doesn't have the higher bound on the dependency.

So I'm okay with this only if we're going to fix the actual issue before the next release, which we should do rather soon.

**Comment by riedgar-ms** [comment]
I would definitely want to release the upper bound again before doing another release. This is _supposed_ to be a quick 'get the builds green' PR (although it appears that `pandas` has other ideas)

**Comment by adrinjalali** [comment]
Releasing with the upper bound doesn't make sense really. As I said, people will end up with the older fairlearn instead. Upper bound only makes sense when every single past release has an upper bound

**Comment by riedgar-ms** [comment]
Sorry, I had too many 'releases' in that sentence, didn't I....

File: fairlearn/metrics/_disaggregated_result.py [hunk 1]
```diff
@@ -39,14 +39,22 @@ def extract_unique_classes(
...
     The incoming DataFrame may have been sliced via `groupby()`.
     This function applies each annotated function in turn to the
     supplied DataFrame.
+
+    The include_groups argument is weird. It appears that pandas
+    introduced it as an argument in v2.2, and immediately deprecated
+    it (dependent on when this is being read, may need to adjust):
+    https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.apply.html
     """
+    assert not include_groups, "Unexpected change to include_groups"
```
Comment (adrinjalali) [inline fairlearn/metrics/_disaggregated_result.py:57]: can we avoid assert in our code? I don't understand the motivation behind this line.

Comment (riedgar-ms) [reply fairlearn/metrics/_disaggregated_result.py:57]: I found that the `include_groups` argument was being included in the invocation of this function. But per `pandas` docs, it should only ever be False. So if it isn't, something really weird is happening; I just wanted to check on this. I can tweak the message

Comment (adrinjalali) [reply fairlearn/metrics/_disaggregated_result.py:57]: but it's not used in this function at all

Comment (riedgar-ms) [reply fairlearn/metrics/_disaggregated_result.py]: Correct. It's needed to work with pandas<2.2 (which I don't think we're dropping yet?). I've removed the `assert` as requested.

**Review by adrinjalali — APPROVED** [review]
at least our tests are now green, let's merge and fix the doc build in a separate PR.

**Pull request merged**